server/main_test.go

```diff
@@ -17,154 +19,137 @@ import (
 	- There's a lot of code repetition here and some shortcuts could be taken like setting the game state directly instead of going through different endpoints
 */
 
+// This function is used for setup before executing the test functions
+func TestMain(m *testing.M) {
+	//Set Gin to Test Mode
+	gin.SetMode(gin.TestMode)
+	// Setup here
+	status := m.Run()
+	// Cleanup here
+	os.Exit(status)
+}
+
 func TestCreateGameRoute(t *testing.T) {
-	router := setupRouter("8080")
-	w := httptest.NewRecorder()
+	// Create the game
 	data := map[string]string{
 		"groupName":  "Kitten Party",
 		"playerName": "Baby Cat",
 	}
-	jsonData, err := json.Marshal(&data)
-	assert.Nil(t, err)
-	req, err := http.NewRequest("POST", "/api/create-game", bytes.NewBuffer(jsonData))
-	assert.Nil(t, err)
-	req.Header.Add("Content-Type", "application/json")
-	router.ServeHTTP(w, req)
-	assert.Equal(t, http.StatusOK, w.Code)
+	req := createRequest(t, "POST", "/api/create-game", data)
+	resp := testHTTPResponse(t, req, http.StatusOK)
+
 	// Todo: Saner expected state
 	expectedState := `{"currentPlayer":{"isHost":true,"name":"Baby Cat"},"currentState":"WaitingForPlayers","groupName":"Kitten Party","players":[{"name":"Baby Cat","host":true,"points":0}]}`
-	assert.Equal(t, expectedState, w.Body.String())
+	assert.Equal(t, expectedState, resp)
```

Comment:
i made a happy object for the response but i think this could still be happier - the dict[string]interface is nice for prototyping but once the interface settles down we might want to use a first class struct with named/typed fields to get some compile time support. not a big issue now but something to think about.

i started looking at using gRPC and google protobufs to do this but that may be a big hammer to hit a tiny nail with.

Reply:
I think json annotations of typed structs would be a good middle ground, kind of the way we parse request bodies like `addPlayerRequest`: The params and types are fixed but validation happens at runtime.

I think that going to protos would slow things down a lot. Definitely open to considering that as an improvement once we have a reasonably stable version :)

Reply:
ack - i agree that we can punt on service interface discussions and in the meantime ill leave this pr as is - we can continue to improve tests as we go.